src/misc/health.rs

```diff
@@ -1,5 +1,64 @@
-use bevy::prelude::warn;
+use bevy::prelude::*;
 use serde::Deserialize;
+use thetawave_interface::spawnable::{EffectType, TextEffectType};
+
+use crate::spawnable::{MobComponent, MobSegmentComponent, SpawnEffectEvent};
+
+#[derive(Event)]
+pub struct DamageDealtEvent {
```

Comment:
Let's move this struct/typedef to `thetawave_interface::health` since many plugins/systems will probably deal with it. But I like using a queue/messaging system.

If we want to do arcade specific things (e.x. with serial ports) when a player takes damage, then this typedef can't be in the main game crate; this typedef (and the corresponding eventReader) should be in a crate imported by the game and the arcade-specific crate.

Reply:
Should this struct/message bus have some notion of what dealt the damage? Maybe some `source: Option<Entity>`? or `reason: Option<Entity>`?

Reply:
I was going to, but I thought you would yagni it